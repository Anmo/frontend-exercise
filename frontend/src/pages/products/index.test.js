import { render } from '@testing-library/react';
import { useRecoilValue } from 'recoil';

import Products from './index';

jest.mock('recoil', () => ({
	...jest.requireActual('recoil'),
	useRecoilValue: jest.fn()
}));

test('renders Products Page', () => {
	useRecoilValue.mockReturnValue(['productId1', 'productId2']);

	const { container } = render(<Products />);

	expect(container.firstChild).toMatchSnapshot();
});

test('renders Products Page with empty list', () => {
	useRecoilValue.mockReturnValue([]);

	const { container } = render(<Products />);

	expect(container.firstChild).toMatchSnapshot();
});
