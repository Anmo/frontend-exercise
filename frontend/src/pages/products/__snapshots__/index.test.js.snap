// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Products Page 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <h1
    class="MuiTypography-root MuiTypography-h5"
  >
    Products
  </h1>
  <div
    class="container"
  >
    <div
      class="MuiPaper-root MuiCard-root container MuiPaper-elevation1 MuiPaper-rounded"
      tabindex="1"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root action"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          />
        </div>
        <div
          class="MuiCardMedia-root media"
          style="background-image: url(/icons/productId1.svg);"
        />
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root container MuiPaper-elevation1 MuiPaper-rounded"
      tabindex="1"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root action"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          />
        </div>
        <div
          class="MuiCardMedia-root media"
          style="background-image: url(/icons/productId2.svg);"
        />
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</main>
`;

exports[`renders Products Page with empty list 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <h1
    class="MuiTypography-root MuiTypography-h5"
  >
    Products
  </h1>
  <div
    class="container"
  />
</main>
`;
